crypto/src/engine/blst/g1.rs

```diff
@@ -0,0 +1,115 @@
+use std::mem::MaybeUninit;
+
+use blst::{
+    blst_p1, blst_p1_affine, blst_p1_affine_compress, blst_p1_affine_in_g1, blst_p1_from_affine,
+    blst_p1_mult, blst_p1_to_affine, blst_p1_uncompress, blst_p1s_to_affine,
+};
+use rayon::prelude::{IntoParallelIterator, ParallelIterator};
+
+use crate::{ParseError, G1};
+
+use super::{scalar::ScalarBlst, BLSTAlgebra};
+
+pub struct G1BlstAffine(MaybeUninit<blst_p1_affine>);
+pub struct G1BlstProjective(MaybeUninit<blst_p1>);
```

Comment:
Can we avoid the `MaybeUninit` here?

Reply:
will try, just helped me handling the raw pointer madness